### PR TITLE
UPGRADE to 0.3.5 => we change the build

### PR DESCRIPTION
**Pull request by jeanlegrais**
UPGRADE to version 0.3.5
This upgrade changes the way we build the package:
-one requirements for the framework s core
-one requirements for the providers and set as extra
-one requirements for the test and set as extra

This upgrade fix the eyereturn issue

### CODE REVIEW

**Review by jinysong — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -9,8 +9,16 @@
 here = os.path.abspath(os.path.dirname(__file__))
 about = {}
 
-with open(os.path.join(here, "requirements.txt"), "r") as f:
-    requirements = f.read().strip().split("\n")
+requirements = ['SQLAlchemy==1.3.23',
```
Comment (eugenemiretsky) [inline setup.py:12]: can we put it in a file?  Similar to requirements.txt? Just another name

File: setup.py [hunk 2]
```diff
@@ -9,8 +9,16 @@
...
+                'apache-airflow-providers-apache-beam==4.0.0',
+                'apache-airflow-providers-common-sql==1.0.0',
+                'apache-airflow-providers-facebook==3.0.1',
```
Comment (eugenemiretsky) [inline setup.py:15]: How come the "apache-airflow-providers-*" are required when packaging, but not required in the main "requirments.txt" file

**Review by eugenemiretsky — CHANGES_REQUESTED** [review]

**Review by eugenemiretsky — COMMENTED** [review]
@jeanlegrais 
We need an explanation of what has been done and why for everybody else

**Comment by eugenemiretsky** [comment]
A couple of overall thoughts: 
1) How did we test it. Ideally you should be able to create a new project, and import the new framework version to make sure it works
2) What are the general guidelines for when to put dependencies in requirements.txt vs inside setup.py  
3) We also wanted to move away from hardcoding a specific version of a lib "==1.0.0" and instead move to requiring a range for versions

File: setup.py (hunk 1 above)
Comment (jeanlegrais) [reply setup.py]: @eugenemiretsky

**Comment by jeanlegrais** [comment]
> A couple of overall thoughts:
> 
> 1. How did we test it. Ideally you should be able to create a new project, and import the new framework version to make sure it works
> 2. What are the general guidelines for when to put dependencies in requirements.txt vs inside setup.py
> 3. We also wanted to move away from hardcoding a specific version of a lib "==1.0.0" and instead move to requiring a range for versions

1. I tested it locally and with Eyereturn project
2. Requirements.txt is meant for local development of the framework
    Install_requires.txt is meant to fit the other dependencies installed with the framework and at the same 
    time limit their scopes
3. Specific versions have been removed from install_requires.txt and are still useful in requirements.txt

File: setup.py (hunk 2 above)
Comment (jeanlegrais) [reply setup.py]: i did change that to put them in requirements.txt instead of install_requires

**Review by jinysong — APPROVED** [review]

**Review by danielkulikov — APPROVED** [review]

**Review by eugenemiretsky — APPROVED** [review]

**Review by eugenemiretsky — COMMENTED** [review]
@jeanlegrais 
requirements-providers.txt. Will it ever be used after Composer fixes it's stuff? 
Maybe a better name is requirements-local.txt or requirements-docker-composer.txt

**Pull request merged**